sdk/search/azure-search-documents/azure/search/documents/_internal/_search_client.py

```diff
@@ -246,25 +246,46 @@ def search(self, search_text, **kwargs):
         select = kwargs.pop("select", None)
         skip = kwargs.pop("skip", None)
         top = kwargs.pop("top", None)
-        query = SearchQuery(
-            search_text=search_text,
-            include_total_result_count=include_total_result_count,
-            facets=facets,
-            filter=filter_arg,
-            highlight_fields=highlight_fields,
-            highlight_post_tag=highlight_post_tag,
-            highlight_pre_tag=highlight_pre_tag,
-            minimum_coverage=minimum_coverage,
-            order_by=order_by,
-            query_type=query_type,
-            scoring_parameters=scoring_parameters,
-            scoring_profile=scoring_profile,
-            search_fields=search_fields,
-            search_mode=search_mode,
-            select=select,
-            skip=skip,
-            top=top
-        )
+        if isinstance(select, list):
+            query = SearchQuery(
+                search_text=search_text,
+                include_total_result_count=include_total_result_count,
+                facets=facets,
+                filter=filter_arg,
+                highlight_fields=highlight_fields,
+                highlight_post_tag=highlight_post_tag,
+                highlight_pre_tag=highlight_pre_tag,
+                minimum_coverage=minimum_coverage,
+                order_by=order_by,
+                query_type=query_type,
+                scoring_parameters=scoring_parameters,
+                scoring_profile=scoring_profile,
+                search_fields=search_fields,
+                search_mode=search_mode,
+                skip=skip,
+                top=top
+            )
+            query.select(select)
+        else:
+            query = SearchQuery(
+                search_text=search_text,
+                include_total_result_count=include_total_result_count,
+                facets=facets,
+                filter=filter_arg,
+                highlight_fields=highlight_fields,
+                highlight_post_tag=highlight_post_tag,
+                highlight_pre_tag=highlight_pre_tag,
+                minimum_coverage=minimum_coverage,
+                order_by=order_by,
+                query_type=query_type,
+                scoring_parameters=scoring_parameters,
+                scoring_profile=scoring_profile,
+                search_fields=search_fields,
+                search_mode=search_mode,
+                select=select,
```

Comment:
can't tell from the code, but is it possible to pass in `select=select if isinstance(select, list) else None` to collapse these two big if else statements?